observability/probe/prober.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package probe
+
+import (
+	"github.com/juju/errors"
+)
+
+// LivenessProber is an interface for probing targets that implement liveness
+// probe support.
+type LivenessProber interface {
+	// LivenessProbe returns the Prober to use for Liveness probes.
+	LivenessProbe() Prober
+}
```

Comment:
We should keep all of these together, ie readiness and startup are below

I wonder, if we are using a map of well known, named probes, could we just have a single Probe() method, not LivenessProbe(), StartupProbe() etc.
Then the aggregator would just iterate over the supplied, named probers and call Probe() on each one

Reply:
Has changed now.